### ISSUE TITLE
breakfix: login_hint and redirect_url are combined in an invalid way

### ISSUE THREAD

**Issue by TobiX** [greenpau/caddy-security#145]
**Describe the issue**

When I'm logged in, but my session times out, on the next request, an invalid auth URL is constructed (query parameters are combined with `?` instead of `&`), loosing the original redirect_url, so the user lands at /auth/portal after re-authentication

**Configuration**

Paste full `Caddyfile` below:

```
{
        auto_https disable_certs
        debug

        security {
                oauth identity provider gitlab {
                        realm gitlab
                        driver gitlab
                        domain_name git.COMPANY.com
                        client_id xxx
                        client_secret yyy
                        scopes openid email profile
                        user_group_filters ^GROUP\d+$
                }

                authentication portal myportal {
                        enable identity provider gitlab
                        cookie domain some.internal.COMPANY.com
                        cookie samesite strict

                        ui {
                                links {
                                        "Service" / icon "las la-search"
                                        "My Identity" /auth/whoami icon "las la-star"
                                }
                        }
                }

                authorization policy mypolicy {
                        set auth url /auth/oauth2/gitlab
                        validate bearer header
                        allow roles git.COMPANY.com/GROUP60
                }
        }
}

some.internal.COMPANY.com {
        tls /etc/ssl/private/cert.pem /etc/ssl/private/key.pem
        root * /home/user/data

        header {
                Strict-Transport-Security "max-age=63072000"
                X-Frame-Options DENY
        }

        route /auth* {
                authenticate with myportal
        }

        route /data* {
                file_server
        }

        route {
                authorize with mypolicy
                reverse_proxy localhost:6070
        }

        handle_errors {
                file_server
                try_files /data/{http.error.status_code}.html /data/{http.error.status_code}.jpg
                respond "{http.error.status_code} {http.error.status_text}"
        }
}
```

**Version Information**

Provide output of `caddy list-modules -versions | grep git` below:

```
http.authentication.hashes.bcrypt v2.5.2
http.authentication.hashes.scrypt v2.5.2
http.authentication.providers.http_basic v2.5.2
http.handlers.authentication v2.5.2
tls.client_auth.leaf v2.5.2
http.authentication.providers.authorizer v1.1.14
http.handlers.authenticator v1.1.14
```

(Note: The grep doesn't work anymore with caddy 2.5.2)

**Expected behavior**

User is redirected back to the URL they came from.

**Additional context**

The flow goes like this:
```
> GET https://some.internal.COMPANY.com/search?q=searchterm+b%3AHEAD&num=150
< 302 https://some.internal.COMPANY.com/auth/oauth2/gitlab/?login_hint=tobias.gruetzmacher%40COMPANY.com?redirect_url=https%3A%2F%2Fsome.internal.COMPANY.com%2Fsearch%3Fq%3Dsearchterm%2Bb%253AHEAD%26num%3D150
> GET https://some.internal.COMPANY.com/auth/oauth2/gitlab/?login_hint=tobias.gruetzmacher%40COMPANY.com?redirect_url=https%3A%2F%2Fsome.internal.COMPANY.com%2Fsearch%3Fq%3Dsearchterm%2Bb%253AHEAD%26num%3D150
< 302 https://git.COMPANY.com/oauth/authorize?client_id=xxx&login_hint=tobias.gruetzmacher%40COMPANY.com%3Fredirect_url%3Dhttps%3A%2F%2Fsome.internal.COMPANY.com%2Fsearch%3Fq%3Dsearchterm%2Bb%253AHEAD%26num%3D150&nonce=XXX&redirect_uri=https%3A%2F%2Fsome.internal.COMPANY.com%2Fauth%2Foauth2%2Fgitlab%2Fauthorization-code-callback&response_type=code&scope=openid+email+profile&state=aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa
```

So it seems the login_hint and redirect_url are combined with `?`, where there should be a `&` ...

**Comment by TobiX** [comment]
Ah, I just found https://github.com/greenpau/go-authcrunch/issues/23, which looks like the same issue...

**Comment by greenpau** [comment]
@TobiX, the fix is now a part of [v1.1.15](https://github.com/greenpau/caddy-security/releases/tag/v1.1.15) release.
